8210927: JDB tests do not update source path after doing a redefine class

alexmenkov opened the pull request:
The fix executes "use" jdb command after class redifinition to improve test logs (print correct source line)

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8210927](https://bugs.openjdk.java.net/browse/JDK-8210927): JDB tests do not update source path after doing a redefine class


### Reviewers
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5664/head:pull/5664` \
`$ git checkout pull/5664`

Update a local copy of the PR: \
`$ git checkout pull/5664` \
`$ git pull https://git.openjdk.java.net/jdk pull/5664/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5664`

View PR using the GUI difftool: \
`$ git pr show -t 5664`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5664.diff">https://git.openjdk.java.net/jdk/pull/5664.diff</a>

</details>

sspitsyn reviewed (approved):
LGTM

plummercj reviewed (commented):
Was there a test case failure, or just something you noticed in the log? If a failure, why were no details ever included in the CR. If just something you noticed in the log, can you include the log output and modify a test to show an error?

Would the synopsis be more accurate if it read "do not update the source _path_ ..."?

alexmenkov commented:
> Was there a test case failure, or just something you noticed in the log? If a failure, why were no details ever included in the CR. If just something you noticed in the log, can you include the log output and modify a test to show an error?

I noticed bad output in test logs.
Added details to the CR.
The test is updated to verify the fix (see uncommented "verifyBPSource(2, bp2Reply);")

> 
> Would the synopsis be more accurate if it read "do not update the source _path_ ..."?

Updated the synopsis.

plummercj commented:
>  The test is updated to verify the fix (see uncommented "verifyBPSource(2, bp2Reply);")

Ah, I assume you added and commented out this test case when you first noticed the problem. Makes sense now.

alexmenkov commented:
> Ah, I assume you added and commented out this test case when you first noticed the problem. Makes sense now.

That's right. Sorry I was not clear enough.

alexmenkov commented:
/integrate
